File: dotnet/cluster_management/CreateMultiRegion.cs
using Amazon;
using Amazon.DSQL;
using Amazon.DSQL.Model;
using Amazon.Runtime;

class MultiRegionClusterCreation {
    public static async Task<CreateMultiRegionClustersResponse> Create(RegionEndpoint region)
    {
        // Create the sdk client
        AWSCredentials aWSCredentials = FallbackCredentialsFactory.GetCredentials();
        AmazonDSQLConfig clientConfig = new()
        {
            AuthenticationServiceName = "dsql",
            RegionEndpoint = region
        };
        AmazonDSQLClient client = new(aWSCredentials, clientConfig);


        // Create multi region cluster
        LinkedClusterProperties USEast1Props = new() {
            DeletionProtectionEnabled = false,
            Tags = new Dictionary<string, string>
            {
                { "Name", "use1-example-cluster" },
                { "Usecase", "testing-mr-use1" }
            }
        };

        LinkedClusterProperties USEast2Props = new() {
            DeletionProtectionEnabled = false,
            Tags = new Dictionary<string, string>
            {
                { "Name", "use2-example-cluster" },
                { "Usecase", "testing-mr-use2" }
            }
        };

        CreateMultiRegionClustersRequest createMultiRegionClustersRequest = new()
        {
            LinkedRegionList = new List<string> { "us-east-1", "us-east-2" },
            WitnessRegion = "us-west-2",
            ClusterProperties = new Dictionary<string, LinkedClusterProperties>
            {
                { "us-east-1", USEast1Props },
                { "us-east-2", USEast2Props }
            }
        };

        CreateMultiRegionClustersResponse createMultiRegionClustersResponse = 
            await client.CreateMultiRegionClustersAsync(createMultiRegionClustersRequest);

        foreach (string arn in createMultiRegionClustersResponse.LinkedClusterArns) {
            Console.WriteLine(arn);
        }

        return createMultiRegionClustersResponse;
    }
}

File: go/cluster_management/create_multi_region.go
package main

import (
	"context"

	dsql "github.com/aws/aws-sdk-go-v2/service/dsql"
	"github.com/aws/aws-sdk-go-v2/service/dsql/types"
)

func CreateMultiRegionCluster(client *dsql.Client) (clustersStatus *dsql.CreateMultiRegionClustersOutput, err error) {

	deleteProtection := false
	witnessRegion := "us-west-2"

	usEast1Props := types.LinkedClusterProperties{
		DeletionProtectionEnabled: &deleteProtection,
		Tags: map[string]string{
			"Name":     "us-east-1-go-example-cluster",
			"Usercase": "testing-mr-use1",
		},
	}

	usEast2Props := types.LinkedClusterProperties{
		DeletionProtectionEnabled: &deleteProtection,
		Tags: map[string]string{
			"Name":     "us-east-2-go-example-cluster",
			"Usercase": "testing-mr-use2",
		},
	}

	clusterProperties := map[string]types.LinkedClusterProperties{
		"us-east-1": usEast1Props,
		"us-east-2": usEast2Props,
	}

	input := dsql.CreateMultiRegionClustersInput{
		LinkedRegionList:  []string{"us-east-1", "us-east-2"},
		WitnessRegion:     &witnessRegion,
		ClusterProperties: clusterProperties,
	}

	clustersStatus, err = client.CreateMultiRegionClusters(context.Background(), &input)
	return
}
